tools/c7n_sphinxext/c7n_sphinxext/docgen.py

```diff
@@ -176,16 +160,38 @@ def _main(provider, output_dir, group_by):
 
     files = []
 
-    # Write out resources by grouped page
-    for key, group in itertools.groupby(
-            sorted(provider_class.resources.values(), key=group_by), key=group_by):
-        rpath = os.path.join(output_dir, "%s.rst" % key)
+    groups = {}
+
+    for r in provider_class.resources.values():
+        group = group_by(r)
+        if not isinstance(group, list):
+            group = [group]
+        for g in group:
+            groups.setdefault(g, []).append(r)
+
+    # Create individual resources pages
+    for r in provider_class.resources.values():
+        rpath = resource_file_name(output_dir, r)
+        with open(rpath, 'w') as fh:
```

Comment:
rather than inlining this code to make this function longer, just reusing the resource directive render function would be nicer.